components/CovidMap.js

```diff
@@ -0,0 +1,702 @@
+      var thirdLegendText = "Federal Facility";
+    } else if(selectedOption == 'age') {
+      var firstLegendText = "Ages Under 35";
+      var secondLegendText = "Ages 35-65";
+      var thirdLegendText = "Ages over 65";
+    } else if(selectedOption == 'ethnicity') {
+      var firstLegendText = "White Deaths";
+      var secondLegendText = "Black Deaths";
+      var thirdLegendText = "Hispanic Deaths";
+    }
+    this.setState({ selectedOption: selectedOption,
+                    selectUpdate: true,
+                    firstLegendText: firstLegendText,
+                    secondLegendText: secondLegendText,
+                    thirdLegendText: thirdLegendText
+                  });
+  }
+
+  render () {
+    const { map, firstClusterer, secondClusterer, thirdClusterer, fetchedMap, data, selectedOption, firstLegendText, secondLegendText, thirdLegendText } = this.state;
+    return (
+      <div id="map-container">
+        <div id="map" style={mapStyle} className="map"></div>
+        <div id="form" style={formStyle} className="form">
+        <form>
+          <div className="form-check">
+            <label>
+              <input type="radio" name="all" value="all" checked={this.state.selectedOption === "all"} onChange={this.handleOptionChange.bind(this)} className="form-check-input"/>
+              All Deaths
+            </label>
+          </div>
+          <div onChange={this.handleOptionChange.bind(this)} className="form-check">
+            <label>
+              <input type="radio" name="facility" value="facility" checked={this.state.selectedOption === "facility"} onChange={this.handleOptionChange.bind(this)} className="form-check-input"/>
+              By Facility
+            </label>
+          </div>
+          <div onChange={this.handleOptionChange.bind(this)} className="form-check">
+            <label>
+              <input type="radio" name="age" value="age" checked={this.state.selectedOption === "age"} onChange={this.handleOptionChange.bind(this)} className="form-check-input"/>
+              By Age
+            </label>
+          </div>
+          { 1 ? null :
+          <div onChange={this.handleOptionChange.bind(this)} className="form-check">
+            <label>
+              <input type="radio" name="ethnicity" value="ethnicity" checked={this.state.selectedOption === "ethnicity"} onChange={this.handleOptionChange.bind(this)} className="form-check-input"/>
+              By Ethnicity
+            </label>
+          </div>
+          }
+        </form>
+        </div>
+        <div id="legend" style={legendStyle} className="legend">
+          <div id="first" style={legendItemStyle} className="legendItem">
+            <div id="legend-first" style={legendTextStyle} className="legendText">{firstLegendText}</div>
+            <div style={legendIconStyle} className="icon"><span style={legendIconFirst} className="legendIcon"></span></div>
+          </div>
+          { (selectedOption == 'all') ? null:
```

Comment:
This bakes in the assumption that every filter other than "all deaths" will have exactly three legend items. Is there a more general approach we could take here?

Reply:
It was a pretty arbitrary decision I had made. For example, I don't know if more or less than three groups would create a better visualization of the spread of ages. But, I think there is a strong case that this can be implemented better by using an array of MarkerClusterer's and could be iterated here to display all groups for our selected filter, however many that may be.